lambdas/functions/block_ips_lambda/app/block_ips.py

```diff
@@ -0,0 +1,239 @@
+import os
+import re
+import time
+from collections import defaultdict
+
+import boto3
+from datetime import datetime, timedelta
+
+# Initialize boto3 clients
+
+table_name = "BlockedIPs"
+ip_set_name = "BlockedIPs"
+ip_set_scope = "REGIONAL"
+
+
+def query_cloudwatch_logs(log_group_name, log_stream_prefix):
+    cloudwatch_logs = boto3.client("logs", region_name="eu-west-1")
+    end_time = datetime.now()
+    # We choose 7 mins so it overlaps the last run (which is every 5 mins)
+    start_time = end_time - timedelta(minutes=7)
+    start_timestamp = int(start_time.timestamp()) * 1000
+    end_timestamp = int(end_time.timestamp()) * 1000
+
+    response = cloudwatch_logs.start_query(
+        logGroupName=log_group_name,
+        startTime=start_timestamp,
+        endTime=end_timestamp,
+        queryString=f"""
+      fields real_forwarded_for, request_uri, status
+      | filter @logStream like "{log_stream_prefix}"
+      | filter request_uri != "/health-check"
+      | filter request_uri != "/login"
+      | filter request_uri != "/"
+      | sort @timestamp desc
+      | limit 10000""",
+    )
+
+    # Get the query ID for fetching results
+    query_id = response["queryId"]
+
+    # Wait for the query to complete
+    while True:
```

Comment:
Should we not include a time out to prevent it running forever?

Reply:
Yeah good call. I'll add that :)